airbyte-integrations/connectors/source-shopify/unit_tests/graphql_bulk/test_job.py

```diff
@@ -167,38 +167,43 @@ def test_job_check(mocker, request, requests_mock, job_response, auth_config, er
             ShopifyBulkExceptions.BulkJobUnknownError, 
             "Could not validate the status of the BULK Job",
         ),
+        (
+            "bulk_successful_response_with_errors", 
+            ShopifyBulkExceptions.BulkJobBadResponse, 
+            "Couldn't check the `response` for `errors`",
+        ),
     ],
     ids=[
-        "success with errors (edge)",
+        "BulkJobUnknownError",
+        "BulkJobBadResponse",
     ],
 )
-def test_one_time_retry_job_check(mocker, request, requests_mock, job_response, auth_config, error_type, expected) -> None:
+def test_retry_on_job_exception(mocker, request, requests_mock, job_response, auth_config, error_type, expected) -> None:
     stream = MetafieldOrders(auth_config)
-    # modify the sleep time for the test
-    stream.job_manager.concurrent_max_retry = 1
-    stream.job_manager.concurrent_interval_sec = 1
-    stream.job_manager.job_check_interval_sec = 1
-    # get job_id from FIXTURE
-    job_id = request.getfixturevalue(job_response).get("data", {}).get("node", {}).get("id")
     # patching the method to get the right ID checks
-    if job_id:
-        mocker.patch("source_shopify.shopify_graphql.bulk.job.ShopifyBulkManager.job_get_id", value=job_id)
+    stream.job_manager._job_id = request.getfixturevalue(job_response).get("data", {}).get("node", {}).get("id")
+    stream.job_manager._job_backoff_time = 0
+    stream.job_manager._job_max_retries = 2
+    # making the check method to raise the specific error
+    mocker.patch(
```

Comment:
`_job_check_for_errors` seems like an implementation detail. Why don't we mock the HTTP request instead?

Reply:
updated.